src/rgw/rgw_rados.cc

```diff
@@ -8007,9 +8017,22 @@ int RGWRados::copy_obj(RGWObjectCtx& obj_ctx,
   attrs.erase(RGW_ATTR_ID_TAG);
   attrs.erase(RGW_ATTR_PG_VER);
   attrs.erase(RGW_ATTR_SOURCE_ZONE);
-  map<string, bufferlist>::iterator cmp = src_attrs.find(RGW_ATTR_COMPRESSION);
-  if (cmp != src_attrs.end())
-    attrs[RGW_ATTR_COMPRESSION] = cmp->second;
+  auto i = src_attrs.find(RGW_ATTR_COMPRESSION);
+  if (i != src_attrs.end())
+    attrs[RGW_ATTR_COMPRESSION] = i->second;
+  // encryption attributes
+  i = src_attrs.find(RGW_ATTR_CRYPT_MODE);
+  if (i != src_attrs.end())
+    attrs[RGW_ATTR_CRYPT_MODE] = i->second;
+  i = src_attrs.find(RGW_ATTR_CRYPT_KEYMD5);
+  if (i != src_attrs.end())
+    attrs[RGW_ATTR_CRYPT_KEYMD5] = i->second;
+  i = src_attrs.find(RGW_ATTR_CRYPT_KEYID);
+  if (i != src_attrs.end())
+    attrs[RGW_ATTR_CRYPT_KEYID] = i->second;
+  i = src_attrs.find(RGW_ATTR_CRYPT_KEYSEL);
+  if (i != src_attrs.end())
+    attrs[RGW_ATTR_CRYPT_KEYSEL] = i->second;
 
```

Comment:
thanks for the tip. it turns out this code wasn't run at all in the `fetch_remote_obj()` path, so i removed that commit